src/main/kotlin/sideChain/iroha/consumer/IrohaConverterImpl.kt

```diff
@@ -19,50 +20,57 @@ class IrohaConverterImpl {
-                            cmd.accountId,
-                            PublicKey(cmd.publicKey)
-                        )
-                    is IrohaCommand.CommandCreateAsset ->
-                        txBuilder = txBuilder.createAsset(
-                            cmd.assetName,
-                            cmd.domainId,
-                            cmd.precision
-                        )
-                    is IrohaCommand.CommandSetAccountDetail ->
-                        txBuilder = txBuilder.setAccountDetail(
-                            cmd.accountId,
-                            cmd.key,
-                            cmd.value
-                        )
-                    is IrohaCommand.CommandTransferAsset ->
-                        txBuilder = txBuilder.transferAsset(
-                            cmd.srcAccountId,
-                            cmd.destAccountId,
-                            cmd.assetId,
-                            cmd.description,
-                            cmd.amount
-                        )
+                for (cmd in transaction.commands) {
+                    when (cmd) {
+                        is IrohaCommand.CommandAddAssetQuantity ->
+                            txBuilder = txBuilder.addAssetQuantity(
+                                cmd.accountId,
+                                cmd.assetId,
+                                cmd.amount
+                            )
+                        is IrohaCommand.CommandAddSignatory ->
+                            txBuilder = txBuilder.addSignatory(
+                                cmd.accountId,
+                                PublicKey(cmd.publicKey)
+                            )
+                        is IrohaCommand.CommandCreateAsset ->
+                            txBuilder = txBuilder.createAsset(
+                                cmd.assetName,
+                                cmd.domainId,
+                                cmd.precision
+                            )
+                        is IrohaCommand.CommandSetAccountDetail ->
+                            txBuilder = txBuilder.setAccountDetail(
+                                cmd.accountId,
+                                cmd.key,
+                                cmd.value
+                            )
+                        is IrohaCommand.CommandTransferAsset ->
+                            txBuilder = txBuilder.transferAsset(
+                                cmd.srcAccountId,
+                                cmd.destAccountId,
+                                cmd.assetId,
+                                cmd.description,
+                                cmd.amount
+                            )
+                    }
                 }
+                txs.add(txBuilder.build())
             }
-            txs.add(txBuilder.build())
+        } catch (e: Exception) {
+            logger.error { e }
         }
         return txs
```

Comment:
I think it is better to return `optional` here.

Reply:
Good point, but it seems better to handle errors with RxJava2. I'll fix it.